src/loaders/genome_attributes/parse_computed_genome_attribs.py

```diff
@@ -0,0 +1,249 @@
+
+        if not docs:
+            raise ValueError(f'Unable to process the computed genome attributes for checkm2 in the specified '
+                             f'directory {batch_dir}.')
+
+        checkm2_docs.extend(docs)
+
+    return checkm2_docs
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='PROTOTYPE - Generate a JSON file for importing into ArangoDB by parsing computed '
+                    'genome attributes.')
+    required = parser.add_argument_group('required named arguments')
+    optional = parser.add_argument_group('optional arguments')
+
+    # Required flag arguments
+    required.add_argument(f'--{loader_common_names.LOAD_VER_ARG_NAME}', required=True, type=str,
+                          help=loader_common_names.LOAD_VER_DESCR)
+
+    required.add_argument(f'--{loader_common_names.KBASE_COLLECTION_ARG_NAME}', required=True, type=str,
+                          help=loader_common_names.KBASE_COLLECTION_DESCR)
+
+    # Optional arguments
+    optional.add_argument('--tools', type=str, nargs='+',
+                          help=f'Extract results from tools. '
+                               f'(default: retrieve all available sub-directories in the '
+                               f'[{loader_common_names.LOAD_VER_ARG_NAME}] directory)')
+    optional.add_argument('--root_dir', type=str, default=loader_common_names.ROOT_DIR,
+                          help=f'Root directory for the collections project. (default: {loader_common_names.ROOT_DIR})')
+    optional.add_argument("-o", "--output", type=argparse.FileType('w'),
+                          help=f"output JSON file path")
+    args = parser.parse_args()
+
+    (tools,
+     load_ver,
+     kbase_collection,
+     root_dir) = (args.tools, getattr(args, loader_common_names.LOAD_VER_ARG_NAME),
+                  getattr(args, loader_common_names.KBASE_COLLECTION_ARG_NAME), args.root_dir)
+
+    result_dir = _locate_dir(root_dir, kbase_collection, load_ver, check_exists=True)
+
+    executed_tools = [d for d in os.listdir(result_dir) if os.path.isdir(os.path.join(result_dir, d))]
+    if not executed_tools:
+        raise ValueError(f'Cannot find any tool result folders in {result_dir}')
+
+    tools = executed_tools if not tools else tools
+    if set(tools) - set(executed_tools):
+        raise ValueError(f'Please ensure that all tools have been successfully executed. '
+                         f'Only the following tools have already been run: {executed_tools}')
+
+    docs = list()
+    for tool in tools:
+        try:
+            parse_ops = getattr(sys.modules[__name__], tool)
+        except AttributeError as e:
+            raise ValueError(f'Please implement parsing method for: [{tool}]') from e
+
+        docs.extend(parse_ops(root_dir, kbase_collection, load_ver))
+
+    docs = merge_docs(docs, '_key')
+
+    output = args.output.name if args.output else '{}_{}_{}'.format(kbase_collection, load_ver,
```

Comment:
Does this leave a file opened by argparse open?

Reply:
right. added code to ensure file is closed at the end.

Reply:
I would just close it immediately since you're not going to use it

Reply:
👍

Reply:
:+1: